### PR TITLE
[dagster-embedded-elt] Fix for mode param not being passed to Sling CLI

### LINKED ISSUE #17301: build_sling_asset doesn't use other SlingMode other than the default FULL_REFRESH
### Dagster version

1.5.3

### What's the issue?

Screenshot 2023-10-19 at 9 34 45 AM

I believe the `_sync` method doesn't currently pass the `mode` to the `config` variable so `build_sling_asset` always use the default SlingMode

### What did you expect to happen?

`mode` to be passed to `_sync` method

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

### PR DESCRIPTION
**Pull request by PedramNavid**
## Summary & Motivation

The mode parameter was not being passed to the Sling CLI resulting in
a FULL_REFRESH on every run.

## How I Tested These Changes

Added a test case

Fixes #17301

### CODE REVIEW

**Comment by PedramNavid** [comment]
Current dependencies on/for this PR:
* master
  * **PR #17310** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/17310" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/17310?utm_source=stack-comment).

File: python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt_tests/test_assets.py [hunk 1]
```diff
@@ -84,3 +84,47 @@ def test_can_build_two_assets():
 
         assert defs.get_assets_def("asset1")
         assert defs.get_assets_def("asset2")
+
+
+def test_sling_modes():
```
Comment (rexledesma) [inline python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt_tests/test_assets.py:89]: Recommend that you parameterize `test_build_sling_asset` with `mode` and `expected_row_counts` after a sling invocation.

**Review by rexledesma — COMMENTED** [review]
Could have less code duplication

File: python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt_tests/test_assets.py (hunk 1 above)
Comment (PedramNavid) [reply python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt_tests/test_assets.py:89]: that make a lot more sense

File: python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt_tests/test_assets.py [hunk 2]
```diff
@@ -2,13 +2,17 @@
...
 from dagster._core.definitions import build_assets_job
 from dagster_embedded_elt.sling import SlingMode, SlingResource, build_sling_asset
 from dagster_embedded_elt.sling.resources import SlingSourceConnection, SlingTargetConnection
 
 
-def test_build_sling_asset():
+@pytest.mark.parametrize(
+    "mode,runs,expected", [(SlingMode.INCREMENTAL, 1, 3), (SlingMode.SNAPSHOT, 2, 6)]
+)
+def test_build_sling_asset(mode, runs, expected):
```
Comment (rexledesma) [inline python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt_tests/test_assets.py:15]: nit: types

**Review by rexledesma — APPROVED** [review]

**Pull request merged**